langchain/llms/base.py

```diff
@@ -287,6 +287,11 @@ def __call__(
         self, prompt: str, stop: Optional[List[str]] = None, callbacks: Callbacks = None
     ) -> str:
         """Check Cache and run the LLM on the given prompt and input."""
+        if isinstance(prompt, list):
```

Comment:
It's common to flip the logic:

```
if not instance(prompt, str):
```

this will then catch a case when `prompt` is a tuple or a set

Reply:
Hi @eyurtsev,  agree, makes more sense that way 👍